tests/test_upload_client.py

```diff
@@ -38,9 +35,39 @@ def parse_graphql_query(query: str) -> Optional[QueryParts]:
         return None
 
     return QueryParts(
-        parts.group("node_type"),
-        parts.group("node_id"),
-        parts.group("fields"),
+        parts.group("node_type"), parts.group("node_id"), parts.group("fields"),
+    )
+
+
+@pytest.fixture
+def mock_simple_upload_client(mock_dir_path):
+    return client.GDCUploadClient(
+        token="dummy",
+        processes=2,
+        server="localhost",
+        upload_part_size=FIVE_MB,
+        multipart=False,
+        files=[
+            {"id": "file-id-2", "path": mock_dir_path},
+            {"id": "file-id-1", "project_id": "GDC-MISC", "path": mock_dir_path},
+        ],
+        debug=False,
+    )
+
+
+@pytest.fixture
+def mock_multipart_upload_client(mock_dir_path):
+    return client.GDCUploadClient(
+        token="dummy",
+        processes=2,
+        server="localhost",
+        upload_part_size=FIVE_MB,
+        multipart=True,
+        files=[
+            {"id": "file-id-2", "path": mock_dir_path},
+            {"id": "file-id-1", "project_id": "GDC-MISC", "path": mock_dir_path},
+        ],
+        debug=False,
     )
```

Comment:
might be nice to use a [partial](https://docs.python.org/3.7/library/functools.html#functools.partial) to freeze the parameters that aren't `multipart` to cut down on duplicate code.

Reply:
i refactored to use a helper function that will return a constructed client based on a passed in boolean for whether it is a multipart or not, I didn't think using partial seemed to apply here, but if the helper function isn't as succinct as using a partial, i can change it.